pkg/storkctl/action.go

```diff
@@ -0,0 +1,116 @@
+package storkctl
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/cmd/util"
+)
+
+const (
+	failoverCommand                    = "failover"
+	nameTimeSuffixFormat string        = "2006-01-02-150405"
+	actionWaitTimeout    time.Duration = 10 * time.Minute
+	actionWaitInterval   time.Duration = 10 * time.Second
+)
+
+func newDoCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	doCommands := &cobra.Command{
+		Use:    "do",
+		Short:  "do actions",
+		Hidden: true,
+	}
+
+	doCommands.AddCommand(
+		newFailoverCommand(cmdFactory, ioStreams),
+	)
+	return doCommands
+}
+
+func newFailoverCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	failoverCommand := &cobra.Command{
+		Use:   failoverCommand,
+		Short: "Initiate failover for the given namespaces",
+		Run: func(c *cobra.Command, args []string) {
+			namespaces, err := cmdFactory.GetAllNamespaces()
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			failed_to_start := false
+			for _, namespace := range namespaces {
+				if incompleteAction := getAnyIncompleteAction(namespace); incompleteAction != nil {
+					if !failed_to_start {
```

Comment:
We don't want promote to be called consecutively on the same volume. This can result in data loss for example when a nearsync replica is added after first promote. This action would be redundant in other cases and best to prevent it?